### PR TITLE
Added hashchange listener

### PR DESCRIPTION
**Pull request by David-OConnor**
Need to make `Url` parsing from string work with hash and search before merging; currently treats them as part of `path`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,9 +2,14 @@
 
 ## v0.4.2
 - Added an `Init` struct, which can help with initial routing (Breaking)
+- Changed `Tag::Custom` to `Tag::from` (Breaking)
```
Comment (MartinKavik) [inline CHANGELOG.md]: I've updated `from` function to accept more input types, no other changes, i.e. no breaking changes.

**Comment by David-OConnor** [comment]
Should now work with hash and search, but is still double-firing; I suspect the `hashchange` event's triggering a `popstate`.

**Comment by MartinKavik** [comment]
@David-OConnor Ping me before merge, please, I would like to do review.

**Comment by David-OConnor** [comment]
Ready now, assuming it works on @flosse's end.

**Comment by flosse** [comment]
> Ready now, assuming it works on @flosse's end.

@David-OConnor thanks a lot, now it works!

**Comment by MartinKavik** [comment]
I think we can merge it so we have this fix  in `master`. We don't have wait for other PRs because of https://github.com/David-OConnor/seed/pull/253#issuecomment-544924580.

**Pull request merged**